conn.go

```diff
@@ -325,43 +346,61 @@ func DialOpen(d Dialer, name string) (_ driver.Conn, err error) {
 		o["user"] = u
 	}
 
-	cn := &conn{
-		opts:   o,
-		dialer: d,
-	}
-	err = cn.handleDriverSettings(o)
+	oo, err := buildHostOptions(o)
 	if err != nil {
 		return nil, err
 	}
-	cn.handlePgpass(o)
+	errs := make(errorSlice, 0)
 
-	cn.c, err = dial(d, o)
-	if err != nil {
-		return nil, err
-	}
+	var cn *conn
+	for _, o = range oo {
+		cn = &conn{
+			opts:   o,
+			dialer: d,
+		}
+		err = cn.handleDriverSettings(o)
+		if err != nil {
+			return nil, err
+		}
+		cn.handlePgpass(o)
 
-	err = cn.ssl(o)
-	if err != nil {
-		return nil, err
-	}
+		cn.c, err = dial(d, o)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
 
-	// cn.startup panics on error. Make sure we don't leak cn.c.
-	panicking := true
-	defer func() {
-		if panicking {
-			cn.c.Close()
+		err = cn.ssl(o)
+		if err != nil {
+			errs = append(errs, err)
+			continue
 		}
-	}()
 
-	cn.buf = bufio.NewReader(cn.c)
-	cn.startup(o)
+		// cn.startup panic on error. Make sure we don't leak cn.c.
+		panicking := true
+		defer func() {
+			if panicking {
+				cn.c.Close()
+			}
+		}()
 
-	// reset the deadline, in case one was set (see dial)
-	if timeout, ok := o["connect_timeout"]; ok && timeout != "0" {
-		err = cn.c.SetDeadline(time.Time{})
+		cn.buf = bufio.NewReader(cn.c)
+		cn.startup(o)
+		if err = cn.checkWritable(o); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		// reset the deadline, in case one was set (see dial)
+		if timeout, ok := o["connect_timeout"]; ok && timeout != "0" {
+			err = cn.c.SetDeadline(time.Time{})
+		}
+		panicking = false
+		return cn, err
 	}
-	panicking = false
-	return cn, err
+
+	// In most cases
```

Comment:
Forgot to finish this sentence.

There's some question about how we would want to return errors/ connections here. Currently this takes the path of least resistance and uses the latest connection / err